### DISCUSSION TITLE
Why nexus can build zkp for  a loop? I think it's a little strange ?

### DISCUSSION THREAD

**Discussion by HappyFox001** [nexus-xyz/nexus-zkvm#330] [General]
why nexus can build a proof some code such as follow:
`#![cfg_attr(target_arch = "riscv32", no_std, no_main)]
use nexus_rt::print;
#[nexus_rt::main]

fn main() {
    let mut a = 0;
    while (true) {
        a += 1;
    }
}
`

**Comment by sjudson** [comment]
Can you explain what you see when you try to prove this program? In my testing, and as expected, the infinite loop causes the tracing/execution of the program to be killed by the OS after running for a length of time, with no proof generated.

**Reply by sjudson** [reply]
Closing due to lack of response, feel free to reopen if you can provide a full example.
